docs/src/implementer_list.md

```diff
@@ -0,0 +1,9 @@
+# Check List for Calculator Implementers
+
+1. Make sure that `AtomsCalculators.promote_force_type` returns correct force type and unit.
+2. Make sure that `AtomsCalculators.zero_forces` returns correct array type and unit for forces.
+3. Non allocating force call has to add to the force input array not overwrite it.
+4. Make sure that your calculator supports nested multithreading. That is it can be called from inside multithreaded loop. This means you should not use static scheduling with `Threads.@threads`.
```

Comment:
I am cocerned here on the upper level multithreading. Main point is that, if you look for exended doc for `Threads.@threads`

```julia
??Threads.@threads
```

Gives you

```julia
:static
  –––––––

  :static scheduler creates one task per thread and divides the iterations
  equally among them, assigning each task specifically to each thread. In
  particular, the value of threadid() is guaranteed to be constant within one
  iteration. Specifying :static is an error if used from inside another
  @threads loop or from a thread other than 1.

  │ Note
  │
  │  :static scheduling exists for supporting transition of code
  │  written before Julia 1.3. In newly written library functions,
  │  :static scheduling is discouraged because the functions using this
  │  option cannot be called from arbitrary worker threads.
```

This means that, if e.g. Molly spawns all calculators in a thread and one of the calculators itself uses static scheduler with `Threads.@threads` the code fails. So, it is possible to implement multithreading in a way that is harmful upstream. We also know there will be improvements to nested multithreading in future Julia version, and use of `:static` is heavily discouraged for the above reason.

Thus we should add this to the interface definition.

As for what this means for MPI, it does not affect it. The only issue is that the main MPI process can have threads doing other calculations, but it does not affect the MPI calculation itself.